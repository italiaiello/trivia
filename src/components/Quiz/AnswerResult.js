import React from 'react'
import CorrectIcon from '../../images/correct.svg'
import WrongIcon from '../../images/wrong.svg'

const AnswerResult = ({     isAnswerCorrect, 
                            nextQuestion, 
                            setIsAnswerCorrect, 
                            chosenAnswer, 
                            correctAnswer 
                }) => {

    const goToNextQuestion = (e) => {
        nextQuestion()
        setIsAnswerCorrect('unanswered')
    }

    return (
        <article className={isAnswerCorrect ? "answerResult correct" : "answerResult wrong"}>
            {
                isAnswerCorrect
                ?
                <article className="resultIconContainer">
                    <figure className="resultIcon">
                        <img src={CorrectIcon} alt="Correct answer icon" />
                    </figure>
                    <p>{`${chosenAnswer} is correct, well done!`}</p>
                </article>
                :
                <article className="resultIconContainer">
                    <figure className="resultIcon">
                        <img src={WrongIcon} alt="Wrong answer icon" />
                    </figure>
                    <p>{`Sorry, ${chosenAnswer} is incorrect.`}</p>
                    <p>{`The correct answer was ${correctAnswer}`}</p>
                </article>
            }
            <button onClick={goToNextQuestion}>Next Question</button>
        </article>
    )
}

export default AnswerResult