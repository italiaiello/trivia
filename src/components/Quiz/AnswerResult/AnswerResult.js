import React from 'react'
import CorrectIcon from '../../../images/correct.svg'
import WrongIcon from '../../../images/wrong.svg'

const AnswerResult = ({     isAnswerCorrect, 
                            nextQuestion, 
                            setIsAnswerCorrect,
                            setHasSelectedAnswer, 
                            chosenAnswer, 
                            correctAnswer,
                            type,
                }) => {

    const goToNextQuestion = (e) => {
        nextQuestion()
        setIsAnswerCorrect('unanswered')
        setHasSelectedAnswer(false)
    }

    return (
        <article className={isAnswerCorrect ? "answerResult correct" : "answerResult wrong"}>
            {
                isAnswerCorrect
                ?
                <article className="resultIconContainer">
                    <figure className="resultIcon">
                        <img src={CorrectIcon} alt="Correct answer icon" />
                    </figure>
                    {
                        type === 'multiple' ?
                        <p>{`${chosenAnswer} is correct, well done!`}</p>
                        :
                        <p>{`That's correct, well done!`}</p>
                    }
                    
                </article>
                :
                <article className="resultIconContainer">
                    <figure className="resultIcon">
                        <img src={WrongIcon} alt="Wrong answer icon" />
                    </figure>
                    {
                        type === 'multiple' ?
                        <article className="wrongResultContainer">
                            <p>{`Sorry, ${chosenAnswer} is incorrect.`}</p>
                            <p>{`The correct answer was ${correctAnswer}`}</p>
                        </article>
                        :
                        <p>{`Sorry, that's incorrect.`}</p>
                    }
                </article>
            }
            <button onClick={goToNextQuestion}>Next Question</button>
        </article>
    )
}

export default AnswerResult