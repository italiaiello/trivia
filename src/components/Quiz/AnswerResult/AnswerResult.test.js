import React from 'react'
import { shallow } from 'enzyme'
import AnswerResult from './AnswerResult'

it('checks if it AnswerResult.js renders properly', () => {

    expect(shallow(<AnswerResult />)).toMatchSnapshot()
})