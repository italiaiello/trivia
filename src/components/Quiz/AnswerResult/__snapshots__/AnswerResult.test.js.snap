// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks if it AnswerResult.js renders properly 1`] = `ShallowWrapper {}`;
